File: api/fetch_gif.py
# api/fetch_gif.py
from flask import Flask, request, send_file
import requests
import os
import re

app = Flask(__name__)

chinese_char_regex = re.compile(r'[\u4e00-\u9fff]')

@app.route('/fetch_gif', methods=['GET'])
def fetch_gif():
    character = request.args.get('char')
    if not chinese_char_regex.match(character):
        return "Invalid character", 400

    gif_dir = '/tmp'  # 使用临时目录
    gif_path = os.path.join(gif_dir, f"{character}.gif")
    
    if not os.path.exists(gif_path):
        # 假设存在一个公开的API或URL可以直接获取GIF
        # 例如: gif_url = f"https://example.com/gifs/{character}.gif"
        # gif_data = requests.get(gif_url).content
        # with open(gif_path, "wb") as f:
        #     f.write(gif_data)
        pass

    return send_file(gif_path, mimetype='image/gif')

# 需要一个启动点，但在Vercel中，你通常不需要这部分代码
if __name__ == "__main__":
    app.run()

File: api/fetch_gif.js
const express = require('express');
const puppeteer = require('puppeteer');
const fs = require('fs');
const path = require('path');

const app = express();
const PORT = process.env.PORT || 3000;

app.get('/fetch_gif', async (req, res) => {
    const { char: character } = req.query;
    if (!character || !/^[\u4e00-\u9fff]+$/.test(character)) {
        return res.status(400).send('Invalid character');
    }

    const gifPath = path.join(__dirname, 'tmp', `${character}.gif`);

    try {
        if (!fs.existsSync(gifPath)) {
            const gifData = await downloadGIF(character);
            if (gifData) {
                fs.writeFileSync(gifPath, gifData, 'binary');
            } else {
                return res.status(404).send('GIF not found');
            }
        }
        res.sendFile(gifPath);
    } catch (error) {
        console.error('Error processing request:', error);
        res.status(500).send('Internal server error');
    }
});

async function downloadGIF(character) {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto('https://hanyu.baidu.com/');

    try {
        await page.type('#kw', character);
        await page.click('#su');
        await page.waitForNavigation();
        await page.waitForSelector('img', { timeout: 10000 });

        const gifUrl = await page.evaluate(() => {
            const img = document.querySelector('img[src*=".gif"]');
            return img ? img.src : null;
        });

        if (gifUrl) {
            const viewSource = await page.goto(gifUrl);
            return await viewSource.buffer();
        }
    } catch (error) {
        console.error('Failed to download GIF:', error);
    } finally {
        await browser.close();
    }
    return null;
}

app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});
